### PR TITLE
Support special characters in username/password

### PR DESCRIPTION
**Pull request by tartale**
- Added unit tests that exhibit usernames/passwords that currently don't work
- Added an alternate connection string parsing that uses tabs instead of ;
  - Alternate parsing does not trim username/password values

### CODE REVIEW

File: mssql.go [hunk 1]
```diff
@@ -125,6 +128,33 @@ func parseConnectionString(dsn string) (res map[string]string) {
...
+			continue
+		}
+		lst := strings.SplitN(part, "=", 2)
+		name := strings.TrimSpace(strings.ToLower(lst[0]))
+		if len(name) == 0 {
+			continue
+		}
+		var value string = ""
+		if len(lst) > 1 {
+			if name != "user id" && name != "password" {
```
Comment (oldenbur) [inline mssql.go]: factor these strings into constants?

**Comment by oldenbur** [comment]
👍 one comment, good coverage

**Review by oldenbur — APPROVED** [review]
one comment, good coverage

File: mssql.go (hunk 1 above)
Comment (tartale) [reply mssql.go]: Good suggestion; done in following commit

File: mssql.go [hunk 2]
```diff
@@ -14,6 +14,11 @@ import (
 	"time"
 )
 
+const (
+	USER_ID_KEY  = "user_id"
+	PASSWORD_KEY = "password"
```
Comment (lcaballero) [inline mssql.go:19]: I thought the key was `User Id` without the '_'.

File: mssql.go [hunk 3]
```diff
@@ -125,6 +133,33 @@ func parseConnectionString(dsn string) (res map[string]string) {
...
+func parseConnectionStringV2(dsn string) (res map[string]string) {
+	res = map[string]string{}
+	parts := strings.Split(dsn, "\t")
+	for _, part := range parts {
+		if len(part) == 0 {
+			continue
+		}
+		lst := strings.SplitN(part, "=", 2)
+		name := strings.TrimSpace(strings.ToLower(lst[0]))
+		if len(name) == 0 {
```
Comment (lcaballero) [inline mssql.go:149]: Would this only happen for a poorly formed string?  You'd have to return an error to indicate as much, so maybe open an issue where the parseConnectionString in general doesn't handle poorly formed string (since it's also in the original)?

File: mssql_test.go [hunk 4]
```diff
@@ -50,3 +50,79 @@ func TestCheckBadConn(t *testing.T) {
...
+		}
+	} else {
+		t.Error("parseConnectionString; result key 'password'; key is missing")
+	}
+}
+
+func TestParseConnectionStringPasswordHasTrailingSpace(t *testing.T) {
+
+	res := parseConnectionString("server=foo\tuser id=bar\tpassword=baz \tencrypt=true\tTrustServerCertificate=true")
+	if val, ok := res["password"]; ok {
```
Comment (lcaballero) [inline mssql_test.go:121]: Need to test `user id` as well for these cases since it's a special case when trimming whitespace.  It likely doesn't work due to the `_` in the reference key.

**Review by lcaballero — CHANGES_REQUESTED** [review]
Need to update with user_id_key tests.

File: mssql.go (hunk 2 above)
Comment (tartale) [reply mssql.go:19]: whoops; I'll fix it in the next commit.  good catch

File: mssql.go (hunk 3 above)
Comment (tartale) [reply mssql.go:149]: I'm assuming that was the purpose; the value would be thrown away in this case.  I'll see if there's an issue, if not I'll create one.

File: mssql_test.go (hunk 4 above)
Comment (tartale) [reply mssql_test.go:121]: Agreed; I'll add it in the next commit

File: mssql.go (hunk 3 above)
Comment (tartale) [reply mssql.go:149]: Added an issue:  https://github.com/denisenkom/go-mssqldb/issues/171
